filters/auth/tokeninfo_test.go

```diff
@@ -15,176 +17,174 @@ import (
 
 const testAuthTimeout = 100 * time.Millisecond
 
+func newOAuthTokeninfoSpec(authType string, options TokeninfoOptions) filters.Spec {
+	switch authType {
+	case filters.OAuthTokeninfoAnyScopeName:
+		return NewOAuthTokeninfoAnyScopeWithOptions(options)
+	case filters.OAuthTokeninfoAllScopeName:
+		return NewOAuthTokeninfoAllScopeWithOptions(options)
+	case filters.OAuthTokeninfoAnyKVName:
+		return NewOAuthTokeninfoAnyKVWithOptions(options)
+	case filters.OAuthTokeninfoAllKVName:
+		return NewOAuthTokeninfoAllKVWithOptions(options)
+	default:
+		panic("unsupported auth type: " + authType)
+	}
+}
+
 func TestOAuth2Tokeninfo(t *testing.T) {
```

Comment:
Refactored this test into two tests: a tokeninfo logic test and invalid arguments test